portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-window.onload = () => { 
+function loadPage() {
+  renderRandomGreeting();
+  renderAuthentication();
+}
+
+function renderAuthentication(){
+  fetch('/authenticate').then((response) => response.json()).then((response) => {
+    debugLog(response);
+    const navbarList = document.getElementById('navbar-list');
+    if(response.isUserLoggedIn){
+      navbarList.appendChild(createAuthenticatedStatusElement(response.logoutUrl, "Logout"));
+    } else {
+      navbarList.appendChild(createAuthenticatedStatusElement(response.loginUrl, "Login"));
+    }
+  });
+}
+
+function createAuthenticatedStatusElement(href, text) {
+  const listElement = document.createElement("li"); 
+  listElement.classList.add("nav-item");
+  listElement.classList.add("active");
+
+  const link = document.createElement("a"); 
+  link.classList.add("nav-link");
+  link.setAttribute("href", href);
+  link.innerHTML = text;
+
+  const span = document.createElement("span");
+  span.classList.add("sr-only");
+  span.innerHTML = "(current)";
+  link.appendChild(span); 
```

Comment:
I'm not sure what this element is for, but it might make more sense if I saw the page. Could you please explain it?

Reply:
I modeled this navbar component off of the other ones that came from Bootstrap. After looking into it, it seems this element is used to hide elements, which isn't needed in my case. l'll remove it and the other ones in the navbar.